### ISSUE TITLE
Fine-tuning got stuck in the beginning

### ISSUE THREAD

**Issue by dian1414** [huggingface/open-r1#331]
I use lora to run stf.py with Bespoke-Stratos-17k. And I use 5 3090s in total, but it seems that the training got stuck as following:

<img width="952" alt="Image" src="https://github.com/user-attachments/assets/5a2d0537-239e-4431-a272-53e70f46d8d6" />

And the gpus still have memory ,

<img width="619" alt="Image" src="https://github.com/user-attachments/assets/cf34d9ae-d1f8-48f6-931c-0bd9761853a8" />

 what leads to that? Attached is my training  parameters and zero3.yaml

CUDA_VISIBLE_DEVICES=3,4,5,6,7 accelerate launch --config_file=recipes/accelerate_configs/zero3.yaml src/open_r1/sft.py \
    --model_name_or_path Qwen/Qwen2___5-1___5B-Instruct \
    --dataset_name HuggingFaceH4/Bespoke-Stratos-17k \
    --learning_rate 2.0e-5 \
    --num_train_epochs 1 \
    --packing \
    --max_seq_length 4096 \
    --per_device_train_batch_size 1 \
    --gradient_accumulation_steps 13 \
    --gradient_checkpointing \
    --bf16 \
    --logging_steps 5 \
    --eval_strategy steps \
    --eval_steps 150 \
    --output_dir data/Qwen2.5-1.5B-Open-R1-Distill

zero3.yaml:
compute_environment: LOCAL_MACHINE
debug: false
deepspeed_config:
  deepspeed_multinode_launcher: standard
  offload_optimizer_device: none
  offload_param_device: none
  zero3_init_flag: true
  zero3_save_16bit_model: true
  zero_stage: 3
distributed_type: DEEPSPEED
downcast_bf16: 'no'
machine_rank: 0
main_training_function: main
mixed_precision: bf16
num_machines: 1
num_processes: 5
rdzv_backend: static
same_network: true
tpu_env: []
tpu_use_cluster: false
tpu_use_sudo: false
use_cpu: false

**Comment by shangshang-wang** [comment]
I got a similar problem today running the GRPO script ... removing the line `peft_config=get_peft_config(model_args)` in the GRPOTrainer works for me.

**Comment by dian1414** [comment]
> 我今天在运行 GRPO 脚本时遇到了类似的问题......删除 GRPOTrainer 中的行对我有用。`peft_config=get_peft_config(model_args)`

Did you use lora or some other peft method ?  Actually I reomved it with my own lora config as follows:

![Image](https://github.com/user-attachments/assets/038b3d18-d2ad-4586-abc6-16fc96310183)
However, it didn't work. Thanks!

**Comment by shangshang-wang** [comment]
Did some more tests on lora:

1. Passing `inference_mode=False`
<img width="504" alt="Image" src="https://github.com/user-attachments/assets/5460eb9d-16af-480f-9ba2-33487d657899" />

2. use `use_reentrant: false`

<img width="322" alt="Image" src="https://github.com/user-attachments/assets/b5c8de21-248a-4069-b7db-45d2ddd2b0af" />

**Comment by dian1414** [comment]
> Did some more tests on lora (probably should not use get_peft_config from trl):
> 
> 1. Passing `inference_mode=False`
> 
> <img alt="Image" width="504" src="https://private-user-images.githubusercontent.com/41543985/413584209-5460eb9d-16af-480f-9ba2-33487d657899.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk2NzMzNjksIm5iZiI6MTczOTY3MzA2OSwicGF0aCI6Ii80MTU0Mzk4NS80MTM1ODQyMDktNTQ2MGViOWQtMTZhZi00ODBmLTliYTItMzM0ODdkNjU3ODk5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjE2VDAyMzEwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI1M2NiYzMxMjJmYTQwZmE5OWVjY2VmNDU3NTVmN2I1MDdhODEwNzRmYmFkOWNkMmVmYzAyZDQxOWQ0YTk5MjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.NBuASnKE-DyjMhBv9doXtNXJzKT9pxYKPRJaTlyF01Q">
> 2. use `use_reentrant: false`
> 
> <img alt="Image" width="322" src="https://private-user-images.githubusercontent.com/41543985/413584233-b5c8de21-248a-4069-b7db-45d2ddd2b0af.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk2NzMzNjksIm5iZiI6MTczOTY3MzA2OSwicGF0aCI6Ii80MTU0Mzk4NS80MTM1ODQyMzMtYjVjOGRlMjEtMjQ4YS00MDY5LWI3ZGItNDVkMmRkZDJiMGFmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjE2VDAyMzEwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVmNzAzZDY2YjcwODBlNzkxMDNmM2RiYjUzNjkyNTc3YjQwZDNjZGUwOWQ4ZTE2ZTY5ZDQ4ZDg2ZjA0NmIzNzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1nlhDBrbmzTmBo7Ow56zzj9VHPYD0zmMtf0meFpXk7c">

Thanks a lot! I'll have a try!

**Comment by ChenDRAG** [comment]
I did not use Lora training but the training still stucks at the beginning. GPU usage is 100% Any ideas?

**Comment by XYaoooo** [comment]
> I did not use Lora training but the training still stucks at the beginning. GPU usage is 100% Any ideas?

same issue

**Comment by shangshang-wang** [comment]
<img width="280" alt="Image" src="https://github.com/user-attachments/assets/547c7dc9-9f57-48b4-85fb-ac78776645b5" />

Maybe turn off vllm in the GRPOTrainer and stick with deepspeed zero2.

**Comment by qgallouedec** [comment]
Please provide the system info (see issue template) including the vllm version

**Comment by XYaoooo** [comment]
Any solution for this, guys ?

**Comment by mgqa34** [comment]
> I did not use Lora training but the training still stucks at the beginning. GPU usage is 100% Any ideas?

The problem i have faced is that the batch sequence length are difference across different devices. Use debug mode can see the errors raised by accelerate. After passing data_collator with padding method "max_length=4096" helps me to solve it.
